t/03-testapi.t

```diff
@@ -334,33 +334,23 @@ subtest 'script_run' => sub {
     is(background_script_run('sleep 10', output => 'foo'), '1234', 'background_script_run with output returns valid PID');
 };
 
-sub assert_script_sudo_test () {
+sub assert_script_sudo_test ($waittime, $is_serial_terminal) {
```

Comment:
Why did you remove the setting of `$script_sudo`?
This test is still passing even if the changes in testapi.pm would be reverted.
It's good that the test checks the return value of `assert_script_sudo`, but that doesn't change with your code changes.
My suggestion from chat was:
```
$mock_testapi->redefine(script_sudo => sub { $script_sudo = $_[0]; return "XXX-0-" })
```
So you can set `$script_sudo` this way and check later that it contains what is expected.

Reply:
You were right. I put back the `is $script_sudo` checks and test it against the old code. Seems to work as expected this time.